utils/ansible_playbook_to_role.py

```diff
@@ -157,7 +165,17 @@ def __init__(self, local_playbook_filename):
     def name(self):
         root, _ = os.path.splitext(os.path.basename(self._local_playbook_filename))
         product, _, profile = root.split("-", 2)
-        return "%s_%s" % (product, profile.replace("-", "_"))
+        return "%s_%s" % (product, profile.replace("-", "_").lower())
+
+    @property
+    @memoize
+    def product(self):
+        return self.name.split("_")[0]
+
+    @property
+    @memoize
+    def profile(self):
+        return self.name.partition("_")[-1]
```

Comment:
Is there a reason to use different functions? (split vs partition)

Reply:
example: rhe7_anssi_nt28_enhanced
```python
>>> name = "rhe7_anssi_nt28_enhanced"
>>> name.split("_")
['rhe7', 'anssi', 'nt28', 'enhanced']
>>> name.partition("_")
('rhe7', '_', 'anssi_nt28_enhanced')
```

partition gives me the actual profile name as the last index, as opposed to chopping together the profile with split(). At least that was my goal with profile, was to get <prod>_[<profile>] as is passed in.

Reply:
@yuumasato my bad, I brainfarted this... so yes, I can use split with the positional arg of 1, and that will give me the second index as my result:
```python
>>> name.split("_", 1)
['rhe7', 'anssi_nt28_enhanced']
```
This way we keep split as the common method of choice...

Reply:
Sorry, I should have given an example use of split when inquiring about it.